src/ansible_runner/cleanup.py

```diff
@@ -152,15 +152,29 @@ def cleanup_dirs(pattern: str, exclude_strings: list | None = None, grace_period
 
 
 def cleanup_images(images: list, runtime: str) -> int:
-    """Note: docker will just untag while podman will remove layers with same command"""
+    """
+    `docker rmi` will just untag while
+    `podman rmi` will untag and remove layers and cause runing container to be killed
+    for podman we use `untag` to achieve the same behavior
+
+    NOTE: this only untag the image and does not delete the image prune_images need to be call to delete
+    """
     rm_ct = 0
     for image_tag in images:
         stdout = run_command([runtime, 'images', '--format="{{.Repository}}:{{.Tag}}"', image_tag])
         if not stdout:
             continue
         for discovered_tag in stdout.split('\n'):
-            stdout = run_command([runtime, 'rmi', discovered_tag.strip().strip('"'), '-f'])
-            rm_ct += stdout.count('Untagged:')
+            if runtime == 'podman':
+                try:
+                    stdout = run_command([runtime, 'untag', image_tag])
+                    if not stdout:
```

Comment:
I don't get it.  Why are we incrementing the count if there is no output?

Reply:
that's what podman does... it doesn't output anything when u run `podman untag` it will however output an error if it fail to untag

Reply:
Why do the condition at all, then?  Rely on the logic getting derailed by the exception.

Reply:
agree with @jbradberry lets keep it simple

Reply:
it shouldn't be needed but it is possible, since this is best effort and not "critical" i rather the behavior to be move on instead of bail

Reply:
How about we get this merged as-in and do an ansible-runner release. Then we can can create a tech-debt task to follow up on this.

Reply:
@gundalow going to have to push back against that.  It's basically a one-line change, and if it's worth doing at all it should be done up-front.  If it merges as-is, it more or less means that we don't care about the possibility of podman's behavior changing and a follow-up task would just sink out of sight.

Reply:
(I'm not opposed to merging as-is, but we shouldn't fool ourselves that we'll ever address such a minor concern unless it bites us in the future.)